src/server/create/microservice/controllers/detail.js

```diff
@@ -7,33 +7,33 @@ import {
   setStepComplete
 } from '~/src/server/create/helpers/form/index.js'
 import { buildErrorDetails } from '~/src/server/common/helpers/build-error-details.js'
-import { fetchServiceTypes } from '~/src/server/create/microservice/helpers/fetch/fetch-service-types.js'
+import { fetchServiceTemplates } from '~/src/server/create/microservice/helpers/fetch/fetch-service-templates.js'
 import { microserviceValidation } from '~/src/server/create/microservice/helpers/schema/microservice-validation.js'
 
 const microserviceDetailController = {
   handler: async (request, h) => {
     const payload = request?.payload
-    const repositoryName = payload.repositoryName
     const serviceTypeTemplate = payload.serviceTypeTemplate
     const teamId = payload.teamId
-    const templateTag = payload.templateTag ?? ''
     const redirectLocation = payload?.redirectLocation
 
-    const { serviceTypes } = await fetchServiceTypes()
-    const serviceTypeTemplates = serviceTypes.map(
-      (serviceType) => serviceType.value
+    const { serviceTemplates } = await fetchServiceTemplates(request)
+    const allowableServiceTemplateNames = serviceTemplates.map(
```

Comment:
So unless the `serviceTemplates` are used. It looks like you are doing this map in two different palces. I'd move it into the fetcher and remove this code